zaza/openstack/charm_tests/glance_simplestreams_sync/tests.py

```diff
@@ -125,3 +125,21 @@ def _check_local_product_streams(expected_images):
         _check_local_product_streams(expected_images)
 
         logging.debug("Local product stream successful")
+
+
+class GlanceSimpleStreamsSyncWithPropertiesTest(GlanceSimpleStreamsSyncTest):
+
+    @classmethod
+    def setUpClass(cls):
+        """Run class setup for running glance simple streams sync tests."""
+        super(GlanceSimpleStreamsSyncWithPropertiesTest, cls).setUpClass()
+        zaza_model.set_application_config('glance-simplestreams-sync', {'set_latest_property': 'true'})
+
+    # TODO(guimalufb) test if the latest property gets removed from old images
+    def test_200_check_image_latest_property(self):
+        """Verify that a property inclusion has been done"""
+        logging.debug("Checking image latest property")
+
+        _filter_properties = {'filters': {'latest': 'true'}}
+        images = self.glance_client.images.list(**_filter_properties)
+        self.assertTrue(len(list(images)) > 0)
```

Comment:
The `len()` function already takes any object that has an `__iter__()` method, so there isn't a need to convert it to a list first.

It's also quite nice to add a message to the `self.assertTrue()` to indicate what the error is in the output if the assertion fails.  e.g. something like "'latest' property not found in glance images list".

Reply:
thanks @ajkavanagh for the quick review! :) 

[`glance_client.images.list` returns a generator](https://docs.openstack.org/python-glanceclient/latest/reference/api/glanceclient.v2.images.html#glanceclient.v2.images.Controller.list), that's why I need to convert it to a list

```python
(Pdb) len(images)
*** TypeError: object of type 'generator' has no len()
```

Reply:
Ah, that's right; generators don't support len.  Tedious.  As it's probably small, `len(list())` is okay.  I'd use `sum(1 for _ in images)` if the generator could produce a lot of items.